### ISSUE TITLE
record time and date of request for CSV export

### ISSUE THREAD

**Issue by StephanJanosch** [chmorgan/chrome-site-load-time#1]
It would be great, if the time and date of the request would be included in the CSV export

**Comment by chmorgan** [comment]
Hey Stephan!

You might be one of the few people using this plugin. I always hoped it
could help someone other than me.

Let me take a look at that. Do you think time and date by default or an
option? By default would be pretty easy I think.

Regards,
Chris

**Comment by StephanJanosch** [comment]
Thanks for replying!

Filtering afterwards is easy, so for my use case request time and date by default would be fine.

I looked at the extension code, but found no clue, how the export works. That's why i could not submit a PR.

Thanks!

**Comment by chmorgan** [comment]
Hi @StephanJanosch what time format would you prefer in the csv?

**Comment by chmorgan** [comment]
Seconds since epoch? iso-8601?

**Comment by chmorgan** [comment]
I don’t blame you, the export code is on a branch and it took me ten
minutes before I figured that out and remembered myself!

I pushed out 0.0.3 of the plugin which is awaiting review by google. It
adds a Unix time field, the first field in fact, to each of the entries in
the csv. Not sure how long it might take to be approved but I’ll email
again when I know more.

Chris

**Comment by StephanJanosch** [comment]
> Seconds since epoch? iso-8601?

I like the current approach. This what I did with the export.

![2020-07-13](https://user-images.githubusercontent.com/11651596/87402139-6b6bbb00-c5bb-11ea-8b55-3a7dfdffdd77.png)

**Comment by chmorgan** [comment]
Nice! Is 0.0.3 working ok for you? If you have any other features you’d
like to see let me know and of course I’d appreciate a review on the
extension if you have a few minutes.

Chris

**Comment by StephanJanosch** [comment]
Yep, 0.0.3 works for me just fine. Except I needed to clean up the 2nd column in the csv export

**Comment by chmorgan** [comment]
What’s up with the second column?

**Comment by StephanJanosch** [comment]
missing quote: https://github.com/chmorgan/chrome-site-load-time/pull/2

**Comment by Reigningfear** [comment]
Hello chmorgan,

This plugin has proven to be very helpful; thank you for your work on this. I did have 1 suggestion for improvement if it's possible. Can we break out the duration_ms column into 3 separate columns to help visualize the network time, server time and client time?

**Comment by chmorgan** [comment]
@Reigningfear this page, https://developers.google.com/web/tools/chrome-devtools/network/understanding-resource-timing, and these are the performance data fields:

connectEnd: 1605751302749 
connectStart: 1605751302715
 domComplete: 1605751303466 
domContentLoadedEventEnd: 1605751303450 
domContentLoadedEventStart: 1605751303450 
domInteractive: 1605751303276 
domLoading: 1605751303235 
domainLookupEnd: 1605751302715
 domainLookupStart: 1605751302709 
fetchStart: 1605751302706 
loadEventEnd: 1605751303469
 loadEventStart: 1605751303466 
navigationStart: 1605751302704 
redirectEnd: 0 
redirectStart: 0 
requestStart: 1605751302749 
responseEnd: 1605751303235 
responseStart: 1605751303218
 secureConnectionStart: 1605751302730 
unloadEventEnd: 1605751303233
 unloadEventStart: 16057513

What fields would you imagine represent network time, server time, and client time?

**Comment by Reigningfear** [comment]
Hello, thanks for the response. To be honest, I am not sure exactly how we would represent that however, I would like to reference another plugin to illustrate the idea - https://chrome.google.com/webstore/detail/simple-performance-bar/gkicgocakpcjjdeigifekmfmpcpepakk. 

In here, this person describes the following to calculate:
_"Network time is connectEnd minus navigationStart.  Server time is responseEnd minus connectEnd.  Finally, client time is loadEventEnd minus responseEnd"_

I was also wondering if there was a way to identify "idle time" as represented in the Chrome Developer tools under "Performance". 

Also, apologies if it's inappropriate to reference another plugin.

**Comment by chmorgan** [comment]
@Reigningfear this is perfect. Is it right? I hope so! It makes sense too. Let me use those numbers and I'll copy that info text into the plugin description as well.

**Comment by chmorgan** [comment]
@Reigningfear I don't see any mention of idle here, https://developers.google.com/web/tools/chrome-devtools/evaluate-performance/performance-reference

**Comment by chmorgan** [comment]
@Reigningfear alright, version 1.0.0 has been pushed out to the store for review. If you get a chance please try it out and let me know what you think!

**Comment by Reigningfear** [comment]
Thank you very much for the quick responses and support! I will definitely try it out within the next few days and get back to you.
